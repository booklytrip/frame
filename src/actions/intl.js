/**
 * @flow
 */

import type { Locale } from '../types';

/**
 * Change current locale to specified
 *
 * @param {String} locale - A new locale to set
 */
export const updateLocale = (locale: Locale) => ({
    type: 'INTL.UPDATE_LOCALE',
    locale,
});
