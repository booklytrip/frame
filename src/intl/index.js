import { addLocaleData } from 'react-intl';
import { reduce, mapValues } from 'lodash';

import en from 'react-intl/locale-data/en';
import lv from 'react-intl/locale-data/lv';
import ru from 'react-intl/locale-data/ru';

import * as defaultLocales from './locales';
import { locales as uikitLocales } from 'uikit';

addLocaleData([
    ...en,
    ...lv,
    ...ru,
]);

/**
 * Combine all messages from list of provides locales
 * and return single object
 */
const combineLocales = (locales) => (
    reduce(locales, (result, locale) => (
        mapValues(locale, (messages, language) => (
            { ...result[language], ...messages }
        ))
    ), {})
);

const locales = combineLocales([defaultLocales, uikitLocales]);

export { locales };
