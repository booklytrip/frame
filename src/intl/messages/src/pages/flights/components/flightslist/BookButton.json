[
  {
    "id": "flights.flightslist.BookButton",
    "defaultMessage": "Book"
  }
]