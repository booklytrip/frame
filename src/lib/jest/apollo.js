import React from 'react';

import { ApolloProvider as Provider } from 'react-apollo';
import ApolloClient from 'apollo-client';
const client = new ApolloClient();

export const ApolloProvider = ({ children }) => (
    <Provider client={ client }>
        { children }
    </Provider>
);
