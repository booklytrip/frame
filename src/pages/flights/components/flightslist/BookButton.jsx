/**
 * @flow
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';
import { darken } from 'polished';

import { Button as BaseButton } from 'semantic-ui-react';

const Button = styled(BaseButton)`
    margin-top: 1em !important;
    margin-right: 0px !important;
    width: 100%;
    text-transform: uppercase !important;
    background: ${({ theme }) =>
        theme.flights.bookButton.background} !important;
    color: ${({ theme }) => theme.flights.bookButton.color} !important;

    &:hover {
        background: ${({ theme }) =>
            darken(0.2, theme.flights.bookButton.background)} !important;
    }
`;

const BookButton = ({ onClick }: { onClick: Function }) =>
    <Button
        onClick={onClick}
        icon="add to cart"
        labelPosition="left"
        content={
            <FormattedMessage
                id="flights.flightslist.BookButton"
                defaultMessage="Book"
            />
        }
    />;

export default BookButton;
