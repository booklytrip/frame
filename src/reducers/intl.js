/**
 * @flow
 */

import { Map } from 'immutable';
import { saveState, loadState } from '../lib/sessionStorage';

const initialState = new Map(loadState('intl') || {});

export default function(state: Object = initialState, action: Object) {
    switch (action.type) {
        // Replace current locale with specified
        case 'INTL.UPDATE_LOCALE': {
            const newState = state.set('locale', action.locale);
            saveState('intl', newState);
            return newState;
        }
        default:
            return state;
    }
}
