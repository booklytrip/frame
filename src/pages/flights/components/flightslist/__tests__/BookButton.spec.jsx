import React from 'react';
import { mountWithIntl, createWithIntl } from '../../../../../lib/jest/intl';
import BookButton from '../BookButton';

describe('BookButton', () => {
    it('renders without crashing', () => {
        mountWithIntl(<BookButton />);
    });

    it('should call onClick event if clicked', () => {
        const spy = jest.fn();
        const bookButton = mountWithIntl(
            <BookButton onClick={ spy } />
        );
        bookButton.simulate('click');

        expect(spy).toBeCalled();
    });

    describe('Snapshots', () => {
        it('renders flight booking button', () => {
            const rendered = createWithIntl(<BookButton />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
