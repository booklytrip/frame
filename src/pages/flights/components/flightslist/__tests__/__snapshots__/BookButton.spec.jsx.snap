// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookButton Snapshots renders flight booking button 1`] = `
<button
  className="ui left labeled yellow icon button iJRTyH"
  onClick={[Function]}
  tabIndex={undefined}
>
  <i
    aria-hidden="true"
    className="add to cart icon"
  />
   
  <span>
    Book
  </span>
</button>
`;
